infrastructure/src/main/java/org/corfudb/infrastructure/DataStore.java

```diff
@@ -109,20 +103,23 @@ public static int getChecksum(byte[] bytes) {
      *
      * @return the cache object
      */
-    private LoadingCache<String, String> buildPersistentDs() {
-        LoadingCache<String, String> cache = Caffeine.newBuilder()
+    private Cache<String, Object> buildPersistentDs() {
+        Cache<String, Object> cache = Caffeine.newBuilder()
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Rename "cache" which hides the field declared at line 55. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AHiddenFieldCheck)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Immediately return this expression instead of assigning it to the temporary variable "cache". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1488)